src/components/functional/PaymentRequestDetailsModal/PaymentRequestDetailsModal.tsx

```diff
@@ -0,0 +1,128 @@
+  apiUrl,
+  merchantId,
+  paymentRequestID,
+  open,
+  onDismiss,
+}: PaymentRequestDetailsModalProps) => {
+  const paymentRequestClient = new PaymentRequestClient(apiUrl, token, merchantId);
+  const merchantClient = new MerchantClient(apiUrl, token, merchantId);
+  const merchantTags = useMerchantTags(apiUrl, token, merchantId);
+  const [localMerchantTags, setLocalMerchantTags] = useState<LocalTag[]>([] as LocalTag[]);
+  const [paymentRequest, setPaymentRequest] = useState<PaymentRequest | undefined>();
+
+  const onRefundClick = async (paymentAttemptID: string) => {
+    //TODO: Will implement refund for atleast card payment attempts. For PISP, it will need to be worked on later.
+    console.log(paymentAttemptID);
+  };
+
+  const onTagAdded = async (tag: LocalTag) => {
+    if (paymentRequest) {
+      const existingTagIds = paymentRequest.tags?.map((tag) => tag.id) ?? [];
+      const paymentRequestUpdate: PaymentRequestUpdate = {
+        tagIds: existingTagIds.concat(tag.id),
+      };
+      const paymentRequestTagAdd = await paymentRequestClient.update(paymentRequest.id, paymentRequestUpdate);
+      if (paymentRequestTagAdd.error) {
+        console.log(paymentRequestTagAdd.error);
+      } else {
+        setPaymentRequest(paymentRequestTagAdd.data);
+      }
+    }
+  };
+
+  const onTagCreated = async (tag: LocalTag) => {
+    if (paymentRequest) {
+      const response = await merchantClient.addTag(parseApiTagToLocalTag(tag));
+      if (response.error) {
+        console.log(response.error);
+      } else {
+        const createdTag = response.data;
+        if (createdTag) {
+          const existingTagIds = paymentRequest.tags?.map((tag) => tag.id) ?? [];
+          const paymentRequestUpdate: PaymentRequestUpdate = {
+            tagIds: existingTagIds.concat(createdTag.id),
+          };
+          const paymentRequestTagAdd = await paymentRequestClient.update(paymentRequest.id, paymentRequestUpdate);
+          if (paymentRequestTagAdd.error) {
+            console.log(paymentRequestTagAdd.error);
+          } else {
+            setPaymentRequest(paymentRequestTagAdd.data);
+          }
+        }
+      }
+    }
+  };
+
+  const onTagDeleted = async (tagIdToDelete: string) => {
+    if (paymentRequest) {
+      const existingTagIds = paymentRequest.tags?.map((tag) => tag.id) ?? [];
+      const paymentRequestUpdate: PaymentRequestUpdate = {
+        tagIds: existingTagIds.filter((id) => id !== tagIdToDelete),
+      };
+      const paymentRequestTagDelete = await paymentRequestClient.update(paymentRequest.id, paymentRequestUpdate);
+      if (paymentRequestTagDelete.error) {
+        console.log(paymentRequestTagDelete.error);
+      } else {
+        setPaymentRequest(paymentRequestTagDelete.data);
+      }
+    }
+  };
+
+  useEffect(() => {
+    if (merchantTags.tags) {
+      setLocalMerchantTags(merchantTags.tags.map((tag) => parseApiTagToLocalTag(tag)));
+    }
+  }, [merchantTags.tags]);
+
+  useEffect(() => {
```

Comment:
There is a custom hook available for this. usePaymentRequest
Also why can't the payment request be passed in to this from the PaymentRequestRow? It seems like when the modal is opened it fetches the payment request when it doesn't need to. This causes a noticeable delay.